numpy/distutils/ccompiler.py

```diff
@@ -136,20 +137,54 @@ def CCompiler_spawn(self, cmd, display=None):
         if is_sequence(display):
             display = ' '.join(list(display))
     log.info(display)
-    s, o = exec_command(cmd)
-    if s:
-        if is_sequence(cmd):
-            cmd = ' '.join(list(cmd))
-        try:
-            print(o)
-        except UnicodeError:
-            # When installing through pip, `o` can contain non-ascii chars
-            pass
-        if re.search('Too many open files', o):
-            msg = '\nTry rerunning setup command until build succeeds.'
-        else:
-            msg = ''
-        raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
+    try:
+        # cmd is documented above to be of type str, but
+        # check_output wants a list -- we use split() to
+        # produce a list irrespective of whether cmd is a single
+        # or multiple commands; historically, exec_command() could
+        # handle string input directly
+        if not is_sequence(cmd):
+            cmd = cmd.split(' ')
+        o = subprocess.check_output(cmd)
+    except subprocess.CalledProcessError as exc:
+        o = exc.output
+        s = exc.returncode
+    except (OSError, AttributeError):
+        # OSError doesn't have the same hooks for the exception
+        # output, but exec_command() historically would use an
+        # empty string for EnvironmentError (base class for
+        # OSError)
+        # NOTE: catching AttributeError is new after migrating
+        # away from exec_command() because if cmd is a sequence
+        # (see control flow below), trying to .split() will raise
+        # an AttributeError
+        o = ''
+        # status previously used by exec_command() for parent
+        # of OSError
+        # NOTE: note sure 127 is appropriate for new AttributeError
+        # though
+        s = 127
+    else:
+        o = filepath_from_subprocess_output(o)
```

Comment:
this line is pointless, I think

Reply:
this isn't reasonable to call unless the goal is actually to get a filename from the thing that was spawned